src/views/SalesAdd.vue

```diff
@@ -0,0 +1,200 @@
+          />
+        </v-container>
+
+        <v-container class="d-flex flex-column mt-4">
+          <p>Total amount: € {{ totalAmount.toFixed(2) || 0 }}</p>
+          <p>Discounted amount: € {{ discountedAmount.toFixed(2) || 0 }}</p>
+          <p>Total discount: {{ totalDiscount.toFixed(2) || 0 }} %</p>
+        </v-container>
+
+        <submit-buttons @reset-form="resetForm" />
+      </v-form>
+    </v-sheet>
+    <products-dialog
+      v-if="selectedUser"
+      v-model="productsDialog"
+      @close-dialog="toggleDialog(false)"
+      @save-product-dialog="productsTableValues"
+      :userId="selectedUser.id"
+    >
+    </products-dialog>
+
+    <calendar-dialog
+      v-model="calendarDialog"
+      @close-dialog="handleCloseCalendar"
+    />
+  </v-container>
+</template>
+
+<script setup>
+import CalendarDialog from "../components/Dialog/CalendarDialog.vue";
+import ProductRow from "../components/ProductRow.vue";
+import { validateAuthors } from "../utils/validation-rules";
+import ProductsDialog from "@/components/Dialog/ProductsDialog.vue";
+import { useRoute, useRouter } from "vue-router";
+import { useStore } from "vuex";
+import { ref, computed, watch, inject } from "vue";
+const snackbarProvider = inject("snackbarProvider");
+const [route, router] = [useRoute(), useRouter()];
+const store = useStore();
+const pageTitle = ref(route.meta.title);
+const [seller, buyer] = [ref([]), ref([])];
+const form = ref(null);
+const selectedUser = ref("");
+const [productsDialog, productsContent] = [ref(false), ref([])];
+const calendarDialog = ref(false);
+const [isProductSelected, isDateSelected] = [ref(false), ref(false)];
+const formattedDate = ref("");
+
+const allUsers = computed(() => store.getters["users/allUsers"]);
+const allUsersNames = allUsers.value.map((user) => user.name);
+
+watch(seller, (newValue) => {
+  selectedUser.value = allUsers.value.find((user) => user.name == seller.value);
+  productsContent.value = [];
+});
+
+const totalAmount = computed(() =>
+  productsContent.value.reduce((acc, cur) => acc + Number(cur.salePrice), 0)
+);
+
+const discountedSmallAmount = computed(() =>
+  productsContent.value.reduce(
+    (acc, cur) => acc + (cur.salePrice * (cur.discount ?? 0)) / 100,
+    0
+  )
+);
+
+const discountedAmount = computed(
+  () => totalAmount.value - discountedSmallAmount.value
+);
+
+const totalDiscount = computed(() => {
+  const totalAmountValue = totalAmount.value;
+  const discountedAmountValue = discountedAmount.value;
+
+  if (totalAmountValue === 0 || isNaN(discountedAmountValue)) {
+    return 0;
+  }
+
+  return 100 - (discountedAmountValue / totalAmountValue) * 100;
+});
+
+const toggleDialog = (isOpen) => {
+  productsDialog.value = isOpen;
+};
+
+function handleCloseCalendar(isOpen, selectedDate) {
+  calendarDialog.value = isOpen;
+  formattedDate.value = selectedDate;
+  isDateSelected.value = true;
+}
+
+const productsTableValues = (productsContentValue) => {
```

Comment:
If I understand it correctly, the parameter here is the selected products for a sale. Let's name them accordingly
```suggestion
const setProductsForSale = (selectedProductsForSale) => {
```

Reply:
fixed